src/rdb.c

```diff
@@ -3075,7 +3075,12 @@ int rdbLoadRioWithLoadingCtx(rio *rdb, int rdbflags, rdbSaveInfo *rsi, rdbLoadin
              * selected data base, in order to avoid useless rehashing. */
             if ((db_size = rdbLoadLen(rdb, NULL)) == RDB_LENERR) goto eoferr;
             if ((expires_size = rdbLoadLen(rdb, NULL)) == RDB_LENERR) goto eoferr;
-            should_expand_db = 1;
+            if (!server.cluster_enabled) {
+                /* Ignore RDB_OPCODE_RESIZEDB in ValKey-8.0 CME: ValKey 8.0 uses dict per slot. Using resize opcode might
+                 * expand all the replica's dictionaries, including dictionaries of unowned slots. This causes memory
+                 * overhead when syncing from ValKey-7 primaries. */
```

Comment:
```suggestion
                /* Ignore RDB_OPCODE_RESIZEDB in Valkey cluster: Since Valkey 8, it uses dictionary per slot. Using resize opcode might
                 * expand all the replica's dictionaries, including dictionaries of unowned slots. This causes memory
                 * overhead when syncing from Valkey 7 primaries. */
```

Reply:
Slightly unclear how we can disregard this opcode for old RDB(s)? We used a heurestic to split the total dbsize across slots owned by the shard and did expansion. Do we just perform incremental rehashing as we load data?